types/errors.go

```diff
@@ -10,3 +11,26 @@ var (
 
 	ErrChallengeAlreadySubmitted = errors.New("challenge is already submitted")
 )
+
+type MultiError struct {
+	errors []error
+}
+
+func NewMultiError(errors []error) *MultiError {
+	return &MultiError{
+		errors: errors,
+	}
+}
+
+func (e *MultiError) Error() string {
+	// concatenate errors
+	err := ""
+	for _, e := range e.errors {
+		if err == "" {
+			err = e.Error()
+		} else {
+			err = fmt.Sprintf("%v | %v", err, e)
+		}
+	}
+	return err
```

Comment:
You can use a `strings.Builder` for this 🙂 

```suggestion
	var b strings.Builder
	b.WriteString(e.errors[0])
	for _, e := range e.errors[1:] {
		b.WriteString(" | ")
		b.WriteString(e)
	}
	return b.String()
```

Reply:
Except it panics when `e.errors` is empty ;)

Reply:
True, but this is never the case.

Why would you return a `MultiErr` that wraps nothing 😉

Reply:
Nothing prevents somebody from:
```go
MultiErr{}.Error()
```
which is a perfectly fine use of its API and would panic. It's not a matter of "why would you return a `MultiErr` that wraps noting" but providing API/implementation that's safe to use.

I will add a length check.